File: 1260-shift-2d-grid/1260-shift-2d-grid.cpp
class Solution {
public:
    void solve(vector<vector<int>> &grid){
        int r = grid.size();
        int c = grid[0].size();

        int fill = grid[0][0];

        for(int i = 0; i < r; i++){
            for(int j = 0; j < c; j++){
                int temp = grid[i][j];
                grid[i][j] = fill;
                fill = temp;
            }
        }

        grid[0][0] = fill;
    }
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        for(int i = 0; i < k; i++){
            solve(grid);
        }

        return grid;
    }
};

File: 1260-shift-2d-grid/1260-shift-2d-grid.py
class Solution:
    def solve(self, grid: list[list[int]]) -> None:
        r = len(grid)
        c = len(grid[0])
        fill = grid[0][0]
        
        for i in range(r):
            for j in range(c):
                temp = grid[i][j]
                grid[i][j] = fill
                fill = temp
                
        grid[0][0] = fill

    def shiftGrid(self, grid: list[list[int]], k: int) -> list[list[int]]:
        for _ in range(k):
            self.solve(grid)
        return grid